src/AzureClient/IAzureClient.cs

```diff
@@ -8,14 +8,119 @@
+        /// </summary>
+        [Description(Resources.AzureClientError_UnknownError)]
+        UnknownError = 0,
+
+        /// <summary>
+        /// No connection has been made to any Azure Quantum workspace.
+        /// </summary>
+        [Description(Resources.AzureClientError_NotConnected)]
+        NotConnected = 1,
+
+        /// <summary>
+        /// A target has not yet been configured for job submission.
+        /// </summary>
+        [Description(Resources.AzureClientError_NoTarget)]
+        NoTarget = 2,
+
+        /// <summary>
+        /// A job meeting the specified criteria was not found.
+        /// </summary>
+        [Description(Resources.AzureClientError_JobNotFound)]
+        JobNotFound = 3,
+
+        /// <summary>
+        /// No Q# operation name was provided where one was required.
+        /// </summary>
+        [Description(Resources.AzureClientError_NoOperationName)]
+        NoOperationName = 4,
+
+        /// <summary>
+        /// Authentication with the Azure service failed.
+        /// </summary>
+        [Description(Resources.AzureClientError_AuthenticationFailed)]
+        AuthenticationFailed = 5,
+
+        /// <summary>
+        /// A workspace meeting the specified criteria was not found.
+        /// </summary>
+        [Description(Resources.AzureClientError_WorkspaceNotFound)]
+        WorkspaceNotFound = 6,
+    }
+
     /// <summary>
     /// This service is capable of connecting to Azure Quantum workspaces
     /// and submitting jobs.
     /// </summary>
     public interface IAzureClient
     {
+        /// <summary>
+        /// Connects to the specified Azure Quantum workspace, first logging into Azure if necessary.
+        /// </summary>
+        public Task<ExecutionResult> ConnectAsync(
+            IChannel channel,
+            string subscriptionId,
+            string resourceGroupName,
+            string workspaceName,
+            string storageAccountConnectionString,
+            bool forceLogin = false);
+
+        /// <summary>
+        /// Prints a string describing the current connection status.
+        /// </summary>
+        public Task<ExecutionResult> PrintConnectionStatusAsync(
+            IChannel channel);
+
+        /// <summary>
+        /// Submits the specified Q# operation as a job to the currently active target.
+        /// </summary>
+        public Task<ExecutionResult> SubmitJobAsync(
+            IChannel channel,
+            IOperationResolver operationResolver,
+            string operationName);
+
+        /// <summary>
+        /// Sets the specified target for job submission.
+        /// </summary>
+        public Task<ExecutionResult> SetActiveTargetAsync(
+            IChannel channel,
+            string targetName);
+        
+        /// <summary>
+        /// Prints the specified target for job submission.
+        /// </summary>
+        public Task<ExecutionResult> PrintActiveTargetAsync(
```

Comment:
wonder why we need an async version of PrintActiveTarget - do we expect this to be blocking?

Reply:
To clarify - Isn't the active target saved locally so this doesn't require a call to the server?

Reply:
Yes, it would be saved locally. I could imagine that maybe we'd also want to do something like retrieve the target's current status (e.g. availability, avg. queue time, etc.).

But in fact, this function isn't even used anywhere at the moment, and there's no scenario in the spec that requires it. I'll just delete it.

---
In reply to: [426993539](https://github.com/microsoft/iqsharp/pull/144#discussion_r426993539) [](ancestors = 426993539)

Reply:
See related comment - deleting this function.

---
In reply to: [426993312](https://github.com/microsoft/iqsharp/pull/144#discussion_r426993312) [](ancestors = 426993312)